inst/tests/froll.Rraw

```diff
@@ -0,0 +1,838 @@
+  c(rep(NA_real_,2), seq(1,1.75,0.25)), c(rep(NA_real_,3), seq(1.125,1.625,0.25))
+)
+test(9999.007, ans1, expected)
+test(9999.008, ans2, expected)
+
+#### in x integer converted to double
+di = data.table(real=1:10/2, int=1:10)
+ans = frollmean(di, 3)
+expected = list(
+  c(rep(NA_real_,2), seq(1,4.5,0.5)),
+  c(rep(NA_real_,2), seq(2,9,1))
+)
+test(9999.009, ans, expected)
+
+#### in n double converted to integer
+x = 1:3/2
+n = 2
+test(9999.010, frollmean(x, n), c(NA, 0.75, 1.25))
+n = list(c(2L,1L,2L), c(2,1,2))
+test(9999.011, frollmean(x, n, adaptive=TRUE), list(c(NA, 1, 1.25), c(NA, 1, 1.25)))
+
+#### error on unsupported type
+dx = data.table(real=1:10/2, char=letters[1:10])
+test(9999.012, frollmean(dx, 3), error="x must be list, data.frame or data.table of numeric types")
+dx = data.table(real=1:10/2, fact=factor(letters[1:10]))
+test(9999.013, frollmean(dx, 3), error="x must be list, data.frame or data.table of numeric types")
+dx = data.table(real=1:10/2, logi=logical(10))
+test(9999.014, frollmean(dx, 3), error="x must be list, data.frame or data.table of numeric types")
+dx = data.table(real=1:10/2, list=rep(list(NA), 10))
+test(9999.015, frollmean(dx, 3), error="x must be list, data.frame or data.table of numeric types")
+x = letters[1:10]
+test(9999.016, frollmean(x, 3), error="x must be of type numeric")
+x = 1:10/2
+test(9999.017, frollmean(x, "a"), error="n must be integer")
+test(9999.018, frollmean(x, factor("a")), error="n must be integer")
+test(9999.019, frollmean(x, TRUE), error="n must be integer")
+test(9999.020, frollmean(x, list(1:10)), error="n must be integer, list is accepted for adaptive TRUE")
+test(9999.021, frollmean(x, list(NA), adaptive=TRUE), error="n must be integer vector or list of integer vectors")
+test(9999.022, frollmean(x, list(c(1:5,1:5), NA), adaptive=TRUE), error="n must be integer vector or list of integer vectors")
+
+#### various length list vectors
+l = list(1:6/2, 3:10/4)
+ans = frollmean(l, c(3, 4))
+expected = list(
+  c(rep(NA_real_,2), seq(1,2.5,0.5)), c(rep(NA_real_,3), seq(1.25,2.25,0.5)),
+  c(rep(NA_real_,2), seq(1,2.25,0.25)), c(rep(NA_real_,3), seq(1.125,2.125,0.25))
+)
+test(9999.023, ans, expected)
+
+#### exact
+set.seed(108)
+x = sample(c(rnorm(1e3, 1e6, 5e5), 5e9, 5e-9))
+n = 15
+ma = function(x, n, na.rm=FALSE) {
+  ans = rep(NA_real_, nx<-length(x))
+  for (i in n:nx) ans[i] = mean(x[(i-n+1):i], na.rm=na.rm)
+  ans
+}
+fastma = function(x, n, na.rm) {
+  if (!missing(na.rm)) stop("NAs are unsupported, wrongly propagated by cumsum")
+  cs = cumsum(x)
+  scs = shift(cs, n)
+  scs[n] = 0
+  as.double((cs-scs)/n)
+}
+ans1 = ma(x, n)
+ans2 = fastma(x, n)
+ans3 = frollmean(x, n, algo="exact")
+ans4 = frollmean(x, n)
+anserr = list(
+  froll_exact_f = ans4-ans1,
+  froll_exact_t = ans3-ans1,
+  fastma = ans2-ans1
+)
+errs = sapply(lapply(anserr, abs), sum, na.rm=TRUE)
+if (.Platform$OS.type=="unix") {
+  test(9999.024, errs[["froll_exact_t"]]==0)
```

Comment:
@mattdowle is it safe to assume exactness on linux? It failed on windows. If we are not sure we could change to `all.equal`, for windows it was not enough.

Reply:
Actually fix didn't resolve windows issue, `exact=T` happens to be less exact than `exact=F`. Will need to investigate that on Windows. We are likely to hit that problem on CRAN as froll tests will kick in there on windows.

Reply:
fixing rounding issue here is not really related to this PR so will be submitted separately to not block this PR